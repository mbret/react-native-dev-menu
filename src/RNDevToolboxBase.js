/**
 * @flow
 */
import React, { Component } from 'react'
import { RNDevToolboxContext } from './RNDevToolboxContext'
import type { Node } from 'react'

type Action = {
  name: string,
  task: Function,
  label?: string
}

export interface RNDevToolboxInterface {
  debug (debug: any): void;
}

export type RNDevToolboxProps = {
  persistenceProvider?: {
    getItem: Function,
    setItem: Function
  },
  onRef?: (ref: RNDevToolboxInterface) => void,
  children?: Node,
  indicators?: any,
  enable?: boolean
}

type State = {
  tipsModalVisible: boolean,
  debug: string,
  indicators: any,
  opened: boolean,
  actions: Array<Action>
}

export class RNDevToolboxBase extends Component<RNDevToolboxProps, State> implements RNDevToolboxInterface {
  componentDidMount () {
    if (this.props.onRef) {
      this.props.onRef(this)
    }
  }

  registerAction = () => {}

  processAction = (name: string) => {}

  debug = (debug: string) => {}

  open = () => {}

  close = () => {}

  toggle = () => {}

  render () {
    return (
      <RNDevToolboxContext.Provider value={this}>
        {this.props.children}
      </RNDevToolboxContext.Provider>
    )
  }
}
