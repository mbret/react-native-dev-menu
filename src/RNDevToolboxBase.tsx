/* eslint no-unused-vars:0 */
/**
 *
 */
import React, { Component } from 'react'
import { Action, Indicator, PersistenceProvider } from './types'

export interface RNDevToolboxInterface {
  debug(debug: any): void;

  processAction(name: string): void;

  registerAction(action: Action | Array<Action>): void;

  open(): void;

  close(): void;

  toggle(): void;
}

export type RNDevToolboxProps<P> = P & {
  persistenceProvider?: PersistenceProvider,
  onRef?: (ref: RNDevToolboxInterface) => void,
  // children?: /* Node */ any,
  indicators?: Array<Indicator>,
  enable?: boolean,
  actions?: Array<Action>
}

export type RNDevToolboxState = {
  tipsModalVisible: boolean,
  debug?: any,
  indicators: Array<Indicator>,
  opened: boolean,
  actions: Array<Action>,
}

export abstract class RNDevToolboxBase<P = {}, S = {}> extends Component<RNDevToolboxProps<P>, S & RNDevToolboxState> implements RNDevToolboxInterface {

  componentDidMount () {
    if (this.props.onRef) {
      this.props.onRef(this)
    }
  }

  processAction (name: string) { }

  registerAction (action: Action | Array<Action>) { }

  debug (debug: any) {
    this.setState({ debug })
  }

  open () {
    // @todo type checking issue here https://github.com/Microsoft/TypeScript/issues/19388
    this.setState<any>({ opened: true })
  }

  close () {
    // @todo type checking issue here https://github.com/Microsoft/TypeScript/issues/19388
    this.setState<any>({ opened: false })
  }

  toggle () {
    this.state.opened ? this.close() : this.open()
  }
}
