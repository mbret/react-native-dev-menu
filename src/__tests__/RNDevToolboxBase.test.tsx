/* eslint-env jest */
/**
 *
 */
import React from 'react'
import { View } from 'react-native'
import { RNDevToolboxBase } from '../RNDevToolboxBase'
import renderer from 'react-test-renderer'

export class InheritBase extends RNDevToolboxBase<any, any> {
  render () {
    return (
      <View />
    )
  }
}

describe('<RNDevToolboxBase>', () => {
  test('renders correctly', () => {
    const a = renderer.create(<InheritBase />).toJSON()
    expect(a).toMatchSnapshot()
  })
})
