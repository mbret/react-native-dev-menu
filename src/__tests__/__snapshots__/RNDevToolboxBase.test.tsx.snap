// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RNDevToolboxBase> renders correctly 1`] = `<View />`;
